scripts/docker/build_docker.py

```diff
@@ -307,39 +529,66 @@ def __parse_arguments(args_list):
     docker_remote_args_group.add_argument('--update-latest', action='store_true',
-                                          help='also update \"latest\" tag in remote docker repo(s)')
-    # flag to turn off git protection (default mode is refusing to build when there are untracked files and/or
-    # uncommitted changes)
+                                          help=f'also update \"{ProjectBuilder.latest_tag}\" tag in remote docker'
+                                               f'repo(s)')
+    dockers_json_path = os.path.join(Paths.gatk_sv_path, "input_values", "dockers.json")
+    docker_remote_args_group.add_argument('--input-json', type=str, default=dockers_json_path,
+                                          help="Path to dockers.json to use as input")
+    docker_remote_args_group.add_argument('--output-json', type=str, default=dockers_json_path,
+                                          help=f"Path to output updated dockers.json. Set to {Paths.dev_null} to turn "
+                                               "off updates")
+    parser.add_argument("--dry-run", action="store_true",
+                        help="Compute docker images that will be build, but don't actually build or push")
+    parser.add_argument("--skip-dependent-images", action="store_true",
+                        help="Don't build images that depend on targets. Can leave images in an unreproducible state: "
+                             "only use this if you are sure you know what you are doing!")
+    parser.add_argument(
+        '--targets', nargs='*', type=str,
+        help='the sub project docker(s) you want to build (note "all" does not include melt)'
+    )
+    short_git_hash_head = get_command_output("git rev-parse --short HEAD").strip()
+    parser.add_argument('--image-tag', type=str, default=short_git_hash_head,
+                        help='tag to be applied to all images being built')
     parser.add_argument(
+        # flag to turn off git protection (default mode is refusing to build when there are untracked files and/or
+        # uncommitted changes)
         '--disable-git-protect', action='store_true',
         help='disable git check/protect when building from local files (will use uncommited changes to build)'
     )
-    parser.add_argument('--skip-base-image-build', action='store_true',
-                        help='skip rebuild of the target\'s base image(s). Assumes that the base image(s) already '
-                             'exist with same tag.')
     parser.add_argument('--no-force-rebuild', action='store_true',
                         help='Do not rebuild docker images if the exact image and tag already exist.')
     parser.add_argument('--skip-cleanup', action='store_true',
                         help='skip cleanup after successful and unsuccessful build attempts.')
+    parser.add_argument('--base-git-commit', type=str,
```

Comment:
The help messages of these two args are clear to me, but I guess they are on the minimal side if you're not familiar with the functionality. For instance, what is `auto-specifying build targets` and what is the role of commit SHAs here?

Reply:
Thanks!